repo2docker/app.py

```diff
@@ -166,29 +183,24 @@ def _user_name_default(self):
 
     def fetch(self, url, ref, checkout_path):
         """Check out a repo using url and ref to the checkout_path location"""
-        try:
-            cmd = ['git', 'clone', '--recursive']
-            if not ref:
-                cmd.extend(['--depth', '1'])
-            cmd.extend([url, checkout_path])
-            for line in execute_cmd(cmd, capture=self.json_logs):
-                self.log.info(line, extra=dict(phase='fetching'))
-        except subprocess.CalledProcessError:
-            self.log.error('Failed to clone repository!',
-                           extra=dict(phase='failed'))
-            sys.exit(1)
-
-        if ref:
-            hash = check_ref(ref, checkout_path)
-            if hash is None:
-                self.log.error('Failed to check out ref %s', ref,
-                               extra=dict(phase='failed'))
-                sys.exit(1)
-            # If the hash is resolved above, we should be able to reset to it
-            for line in execute_cmd(['git', 'reset', '--hard', hash],
-                                    cwd=checkout_path,
-                                    capture=self.json_logs):
-                self.log.info(line, extra=dict(phase='fetching'))
+        # Pick a content provider based on URL
```

Comment:
Perhaps this comment is better in the docstring. Or modify to something like: Iterate through possible content providers until a valid provider, based on URL, is found.